In [1]:
from mlp_dropout_dropconnect import *


Using gpu device 0: GRID K520


In [2]:
# No-drop on linear
test_mlp(learning_rate = 0.1, momentum = 0.5, L1_reg= 0.00, L2_reg= 0.0001, n_epochs=100, batch_size=20, n_hidden=500, n_hiddenLayers=2, hidden_layer_type= DropoutHiddenLayer,p = 1, dataset = 'MNIST', verbose = True, smaller_set = False,filename = 'mlp_original_mnist_linear',activation = linear_rectifier)

train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 5.410000 %
     epoch 1, minibatch 2500/2500, test error of best model 5.900000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 4.530000 %
     epoch 2, minibatch 2500/2500, test error of best model 4.810000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 3.330000 %
     epoch 3, minibatch 2500/2500, test error of best model 3.390000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 3.260000 %
     epoch 4, minibatch 2500/2500, test error of best model 3.240000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatch 2500

In [3]:
# No-drop on tanh
test_mlp(learning_rate = 0.1, momentum = 0.5, L1_reg= 0.00, L2_reg= 0.0001, n_epochs=100, batch_size=20, n_hidden=500, n_hiddenLayers=2, hidden_layer_type= DropoutHiddenLayer,p = 1, dataset = 'MNIST', verbose = True, smaller_set = False,filename = 'mlp_original_mnist_tanh',activation = T.tanh)

train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 8.660000 %
     epoch 1, minibatch 2500/2500, test error of best model 9.830000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 5.410000 %
     epoch 2, minibatch 2500/2500, test error of best model 5.800000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 3.970000 %
     epoch 3, minibatch 2500/2500, test error of best model 4.310000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 3.430000 %
     epoch 4, minibatch 2500/2500, test error of best model 3.620000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatch 2500

In [4]:
#No-drop on sigmoid
test_mlp(learning_rate = 0.1, momentum = 0.5, L1_reg= 0.00, L2_reg= 0.0001, n_epochs=100, batch_size=20, n_hidden=500, n_hiddenLayers=2, hidden_layer_type= DropoutHiddenLayer,p = 1, dataset = 'MNIST', verbose = True, smaller_set = False,filename = 'mlp_original_mnist_sigmoid',activation = theano.tensor.nnet.sigmoid)

train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 9.530000 %
     epoch 1, minibatch 2500/2500, test error of best model 10.390000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 8.110000 %
     epoch 2, minibatch 2500/2500, test error of best model 8.990000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 6.780000 %
     epoch 3, minibatch 2500/2500, test error of best model 7.720000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 5.940000 %
     epoch 4, minibatch 2500/2500, test error of best model 6.920000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatch 250

In [5]:
# Dropout with changing hidden units
for hidden_units in [200,400,800,1600]:
    test_mlp(learning_rate = 0.1, momentum = 0.5, L1_reg= 0.00, L2_reg= 0.0001, n_epochs=100, batch_size=20, n_hidden=hidden_units, n_hiddenLayers=2, hidden_layer_type= DropoutHiddenLayer,p = 0.5, dataset = 'MNIST', verbose = True, smaller_set = False,filename = 'mlp_original_mnist_linear%s'%hidden_units,activation = linear_rectifier)
    

train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 6.850000 %
     epoch 1, minibatch 2500/2500, test error of best model 7.500000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 4.580000 %
     epoch 2, minibatch 2500/2500, test error of best model 5.000000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 4.130000 %
     epoch 3, minibatch 2500/2500, test error of best model 4.580000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 3.500000 %
     epoch 4, minibatch 2500/2500, test error of best model 3.920000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatch 2500

In [7]:
 # DROPOUT with linear 
test_mlp(learning_rate = 0.1, momentum = 0.5, L1_reg= 0.00, L2_reg= 0.0001, n_epochs=100, batch_size=20, n_hidden=500, n_hiddenLayers=2, hidden_layer_type= DropoutHiddenLayer,p = 0.5, dataset = 'MNIST', verbose = True, smaller_set = False,filename = 'mlp_dropout_mnist_linear',activation = linear_rectifier)

train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 6.060000 %
     epoch 1, minibatch 2500/2500, test error of best model 6.360000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 3.860000 %
     epoch 2, minibatch 2500/2500, test error of best model 4.610000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 3.470000 %
     epoch 3, minibatch 2500/2500, test error of best model 3.780000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 2.910000 %
     epoch 4, minibatch 2500/2500, test error of best model 3.260000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatch 2500

In [8]:
#Dropout with tanh
test_mlp(learning_rate = 0.1, momentum = 0.5, L1_reg= 0.00, L2_reg= 0.0001, n_epochs=100, batch_size=20, n_hidden=500, n_hiddenLayers=2, hidden_layer_type= DropoutHiddenLayer,p = 0.5, dataset = 'MNIST', verbose = True, smaller_set = False,filename = 'mlp_dropout_mnist_tanh',activation = T.tanh)

train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 8.360000 %
     epoch 1, minibatch 2500/2500, test error of best model 8.720000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 7.310000 %
     epoch 2, minibatch 2500/2500, test error of best model 7.460000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 5.790000 %
     epoch 3, minibatch 2500/2500, test error of best model 6.160000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 5.220000 %
     epoch 4, minibatch 2500/2500, test error of best model 5.650000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatch 2500

In [10]:
#Dropout with sigmoid 
test_mlp(learning_rate = 0.1, momentum = 0.5, L1_reg= 0.00, L2_reg= 0.0001, n_epochs=100, batch_size=20, n_hidden=500, n_hiddenLayers=2, hidden_layer_type= DropoutHiddenLayer,p = 0.5, dataset = 'MNIST', verbose = True, smaller_set = False,filename = 'mlp_dropout_mnist_sigmoid',activation = theano.tensor.nnet.sigmoid)

train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 13.620000 %
     epoch 1, minibatch 2500/2500, test error of best model 14.180000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 11.220000 %
     epoch 2, minibatch 2500/2500, test error of best model 11.590000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 10.000000 %
     epoch 3, minibatch 2500/2500, test error of best model 10.520000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 9.460000 %
     epoch 4, minibatch 2500/2500, test error of best model 9.900000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatc

In [11]:
#Normal MLP changing hidden layers
for hidden_units in [200,400,800,1600]:
    test_mlp(learning_rate = 0.1, momentum = 0.5, L1_reg= 0.00, L2_reg= 0.0001, n_epochs=100, batch_size=20, n_hidden=hidden_units, n_hiddenLayers=2, hidden_layer_type= DropoutHiddenLayer,p = 1, dataset = 'MNIST', verbose = True, smaller_set = False,filename = 'mlp_original_mnist_linear%s'%hidden_units,activation = linear_rectifier)


train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 5.310000 %
     epoch 1, minibatch 2500/2500, test error of best model 6.290000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 4.670000 %
     epoch 2, minibatch 2500/2500, test error of best model 4.990000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 4.210000 %
     epoch 3, minibatch 2500/2500, test error of best model 4.230000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 3.130000 %
     epoch 4, minibatch 2500/2500, test error of best model 3.020000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatch 2500

In [ ]:
#Dropout with different p values below.

In [ ]:
for p in [0.1,0.2,0.3,0.4,0.5]:
    test_mlp(learning_rate = 0.1, momentum = 0.5, L1_reg= 0.00, L2_reg= 0.0001, n_epochs=100, batch_size=20, n_hidden=800, n_hiddenLayers=2, hidden_layer_type= DropoutHiddenLayer,p = p, dataset = 'MNIST', verbose = True, smaller_set = False,filename = 'mlp_dropout_mnist_linear_%s'%p,activation = linear_rectifier)

train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 90.090000 %
     epoch 1, minibatch 2500/2500, test error of best model 90.200000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 89.750000 %
     epoch 2, minibatch 2500/2500, test error of best model 89.750000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 86.750000 %
     epoch 3, minibatch 2500/2500, test error of best model 86.880000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 71.510000 %
     epoch 4, minibatch 2500/2500, test error of best model 71.860000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatch 2500/2500, validation error 54.570000 %
     epoch 5, minibatch 2500/2500, test error of best model 55.600000 %
momentum:  0.548039615154
learning rate:  0.0927216502366
epoch 6, minibatch 2500/2500, validation error 48.970000 %
     epoch 6, minibatch 2500/2500, test error of best model 51.020000 %
momentum:  0.557078838348
learning rate:  0.091330825483
epoch 7, minibatch 2500/2500, validation error 41.360000 %
     epoch 7, minibatch 2500/2500, test error of best model 43.120000 %
momentum:  0.565937280655
learning rate:  0.0899608631008
epoch 8, minibatch 2500/2500, validation error 37.950000 %
     epoch 8, minibatch 2500/2500, test error of best model 39.080000 %
momentum:  0.574618518353
learning rate:  0.0886114501543
epoch 9, minibatch 2500/2500, validation error 36.420000 %
     epoch 9, minibatch 2500/2500, test error of best model 37.460000 %
momentum:  0.58312612772
learning rate:  0.0872822784019
epoch 10, minibatch 2500/2500, validation error 33.060000 %
     epoch 10, minibatch 2500/2500, test error of best model 34.040000 %
momentum:  0.591463625431
learning rate:  0.0859730442259
epoch 11, minibatch 2500/2500, validation error 31.520000 %
     epoch 11, minibatch 2500/2500, test error of best model 32.650000 %
momentum:  0.599634349346
learning rate:  0.0846834485625
epoch 12, minibatch 2500/2500, validation error 28.220000 %
     epoch 12, minibatch 2500/2500, test error of best model 28.990000 %
momentum:  0.607641637325
learning rate:  0.0834131968341
epoch 13, minibatch 2500/2500, validation error 27.790000 %
     epoch 13, minibatch 2500/2500, test error of best model 28.630000 %
momentum:  0.615488827229
learning rate:  0.0821619988816
epoch 14, minibatch 2500/2500, validation error 24.970000 %
     epoch 14, minibatch 2500/2500, test error of best model 26.170000 %
momentum:  0.623179078102
learning rate:  0.0809295688984
epoch 15, minibatch 2500/2500, validation error 24.770000 %
     epoch 15, minibatch 2500/2500, test error of best model 25.860000 %
momentum:  0.630715489388
learning rate:  0.0797156253649
epoch 16, minibatch 2500/2500, validation error 24.000000 %
     epoch 16, minibatch 2500/2500, test error of best model 25.150000 %
momentum:  0.638101160526
learning rate:  0.0785198909844
epoch 17, minibatch 2500/2500, validation error 23.450000 %
     epoch 17, minibatch 2500/2500, test error of best model 24.730000 %
momentum:  0.645339131355
learning rate:  0.0773420926196
epoch 18, minibatch 2500/2500, validation error 23.140000 %
     epoch 18, minibatch 2500/2500, test error of best model 24.520000 %
momentum:  0.652432322502
learning rate:  0.0761819612303
epoch 19, minibatch 2500/2500, validation error 22.870000 %
     epoch 19, minibatch 2500/2500, test error of best model 24.070000 %
momentum:  0.659383654594
learning rate:  0.0750392318119
epoch 20, minibatch 2500/2500, validation error 22.080000 %
     epoch 20, minibatch 2500/2500, test error of best model 23.310000 %
momentum:  0.666195988655
learning rate:  0.0739136433347
epoch 21, minibatch 2500/2500, validation error 21.220000 %
     epoch 21, minibatch 2500/2500, test error of best model 22.500000 %
momentum:  0.672872066498
learning rate:  0.0728049386847
epoch 22, minibatch 2500/2500, validation error 21.160000 %
     epoch 22, minibatch 2500/2500, test error of best model 22.380000 %
momentum:  0.679414629936
learning rate:  0.0717128646044
epoch 23, minibatch 2500/2500, validation error 20.870000 %
     epoch 23, minibatch 2500/2500, test error of best model 22.150000 %
momentum:  0.685826361179
learning rate:  0.0706371716354
epoch 24, minibatch 2500/2500, validation error 20.000000 %
     epoch 24, minibatch 2500/2500, test error of best model 21.400000 %
momentum:  0.692109823227
learning rate:  0.0695776140608
epoch 25, minibatch 2500/2500, validation error 19.980000 %
     epoch 25, minibatch 2500/2500, test error of best model 21.260000 %
momentum:  0.698267638683
learning rate:  0.0685339498499
epoch 26, minibatch 2500/2500, validation error 19.850000 %
     epoch 26, minibatch 2500/2500, test error of best model 21.130000 %
momentum:  0.704302310944
learning rate:  0.0675059406022
epoch 27, minibatch 2500/2500, validation error 19.380000 %
     epoch 27, minibatch 2500/2500, test error of best model 20.680000 %
momentum:  0.710216283798
learning rate:  0.0664933514931
epoch 28, minibatch 2500/2500, validation error 19.930000 %
momentum:  0.716011941433
learning rate:  0.0654959512207
epoch 29, minibatch 2500/2500, validation error 19.400000 %
momentum:  0.721691727638
learning rate:  0.0645135119524
epoch 30, minibatch 2500/2500, validation error 19.390000 %
momentum:  0.727257907391
learning rate:  0.0635458092731
epoch 31, minibatch 2500/2500, validation error 18.930000 %
     epoch 31, minibatch 2500/2500, test error of best model 20.170000 %
momentum:  0.732712745667
learning rate:  0.062592622134
epoch 32, minibatch 2500/2500, validation error 18.890000 %
     epoch 32, minibatch 2500/2500, test error of best model 20.150000 %
momentum:  0.738058507442
learning rate:  0.061653732802
epoch 33, minibatch 2500/2500, validation error 18.450000 %
     epoch 33, minibatch 2500/2500, test error of best model 19.860000 %
momentum:  0.743297338486
learning rate:  0.06072892681
epoch 34, minibatch 2500/2500, validation error 18.510000 %
momentum:  0.748431384563
learning rate:  0.0598179929078
epoch 35, minibatch 2500/2500, validation error 18.690000 %
momentum:  0.753462731838
learning rate:  0.0589207230142
epoch 36, minibatch 2500/2500, validation error 18.210000 %
     epoch 36, minibatch 2500/2500, test error of best model 19.400000 %
momentum:  0.758393466473
learning rate:  0.058036912169
epoch 37, minibatch 2500/2500, validation error 18.410000 %
momentum:  0.763225615025
learning rate:  0.0571663584865
epoch 38, minibatch 2500/2500, validation error 18.220000 %
momentum:  0.767961084843
learning rate:  0.0563088631092
epoch 39, minibatch 2500/2500, validation error 18.200000 %
     epoch 39, minibatch 2500/2500, test error of best model 19.620000 %
momentum:  0.77260184288
learning rate:  0.0554642301625
epoch 40, minibatch 2500/2500, validation error 18.230000 %
momentum:  0.777149796486
learning rate:  0.0546322667101
epoch 41, minibatch 2500/2500, validation error 17.960000 %
     epoch 41, minibatch 2500/2500, test error of best model 19.380000 %
momentum:  0.781606793404
learning rate:  0.0538127827095
epoch 42, minibatch 2500/2500, validation error 17.900000 %
     epoch 42, minibatch 2500/2500, test error of best model 19.200000 %
momentum:  0.785974681377
learning rate:  0.0530055909688
epoch 43, minibatch 2500/2500, validation error 17.910000 %
momentum:  0.790255188942
learning rate:  0.0522105071043
epoch 44, minibatch 2500/2500, validation error 17.460000 %
     epoch 44, minibatch 2500/2500, test error of best model 18.680000 %
momentum:  0.794450104237
learning rate:  0.0514273494977
epoch 45, minibatch 2500/2500, validation error 17.450000 %
     epoch 45, minibatch 2500/2500, test error of best model 18.520000 %
momentum:  0.798561096191
learning rate:  0.0506559392553
epoch 46, minibatch 2500/2500, validation error 17.870000 %
momentum:  0.802589893341
learning rate:  0.0498961001664
epoch 47, minibatch 2500/2500, validation error 17.630000 %
momentum:  0.806538105011
learning rate:  0.0491476586639
epoch 48, minibatch 2500/2500, validation error 17.670000 %
momentum:  0.810407340527
learning rate:  0.048410443784
epoch 49, minibatch 2500/2500, validation error 17.820000 %
momentum:  0.814199209213
learning rate:  0.0476842871272
epoch 50, minibatch 2500/2500, validation error 17.640000 %
momentum:  0.817915201187
learning rate:  0.0469690228203
epoch 51, minibatch 2500/2500, validation error 17.480000 %
momentum:  0.821556925774
learning rate:  0.046264487478
epoch 52, minibatch 2500/2500, validation error 17.780000 %
momentum:  0.825125813484
learning rate:  0.0455705201658
epoch 53, minibatch 2500/2500, validation error 17.500000 %
momentum:  0.82862329483
learning rate:  0.0448869623633
epoch 54, minibatch 2500/2500, validation error 17.270000 %
     epoch 54, minibatch 2500/2500, test error of best model 18.380000 %
momentum:  0.832050800323
learning rate:  0.0442136579279
epoch 55, minibatch 2500/2500, validation error 17.470000 %
momentum:  0.835409760475
learning rate:  0.043550453059
epoch 56, minibatch 2500/2500, validation error 17.310000 %
momentum:  0.838701546192
learning rate:  0.0428971962631
epoch 57, minibatch 2500/2500, validation error 17.430000 %
momentum:  0.841927528381
learning rate:  0.0422537383191
epoch 58, minibatch 2500/2500, validation error 17.470000 %
momentum:  0.84508895874
learning rate:  0.0416199322444
epoch 59, minibatch 2500/2500, validation error 17.190000 %
     epoch 59, minibatch 2500/2500, test error of best model 18.360000 %
momentum:  0.848187208176
learning rate:  0.0409956332607
epoch 60, minibatch 2500/2500, validation error 17.360000 %
momentum:  0.851223468781
learning rate:  0.0403806987618
epoch 61, minibatch 2500/2500, validation error 17.070000 %
     epoch 61, minibatch 2500/2500, test error of best model 18.270000 %
momentum:  0.854198992252
learning rate:  0.0397749882803
epoch 62, minibatch 2500/2500, validation error 17.050000 %
     epoch 62, minibatch 2500/2500, test error of best model 18.260000 %
momentum:  0.857115030289
learning rate:  0.0391783634561
epoch 63, minibatch 2500/2500, validation error 17.260000 %
momentum:  0.859972715378
learning rate:  0.0385906880043
epoch 64, minibatch 2500/2500, validation error 17.070000 %
momentum:  0.862773239613
learning rate:  0.0380118276842
epoch 65, minibatch 2500/2500, validation error 16.990000 %
     epoch 65, minibatch 2500/2500, test error of best model 18.210000 %
momentum:  0.865517795086
learning rate:  0.037441650269
epoch 66, minibatch 2500/2500, validation error 17.000000 %
momentum:  0.868207454681
learning rate:  0.0368800255149
epoch 67, minibatch 2500/2500, validation error 16.960000 %
     epoch 67, minibatch 2500/2500, test error of best model 18.090000 %
momentum:  0.870843291283
learning rate:  0.0363268251322
epoch 68, minibatch 2500/2500, validation error 17.130000 %
momentum:  0.873426437378
learning rate:  0.0357819227552
epoch 69, minibatch 2500/2500, validation error 17.330000 %
momentum:  0.875957906246
learning rate:  0.0352451939139
epoch 70, minibatch 2500/2500, validation error 16.630000 %
     epoch 70, minibatch 2500/2500, test error of best model 17.940000 %
momentum:  0.878438770771
learning rate:  0.0347165160052
epoch 71, minibatch 2500/2500, validation error 16.950000 %
momentum:  0.880869984627
learning rate:  0.0341957682651
epoch 72, minibatch 2500/2500, validation error 17.030000 %
momentum:  0.883252561092
learning rate:  0.0336828317411
epoch 73, minibatch 2500/2500, validation error 16.900000 %
momentum:  0.885587513447
learning rate:  0.033177589265
epoch 74, minibatch 2500/2500, validation error 16.940000 %
momentum:  0.88787573576
learning rate:  0.032679925426
epoch 75, minibatch 2500/2500, validation error 16.840000 %
momentum:  0.89011824131
learning rate:  0.0321897265447
epoch 76, minibatch 2500/2500, validation error 17.050000 %
momentum:  0.892315864563
learning rate:  0.0317068806465
epoch 77, minibatch 2500/2500, validation error 16.880000 %
momentum:  0.894469559193
learning rate:  0.0312312774368
epoch 78, minibatch 2500/2500, validation error 16.870000 %
momentum:  0.896580159664
learning rate:  0.0307628082752
epoch 79, minibatch 2500/2500, validation error 16.800000 %
momentum:  0.898648560047
learning rate:  0.0303013661511
epoch 80, minibatch 2500/2500, validation error 17.040000 %
momentum:  0.900675594807
learning rate:  0.0298468456588
epoch 81, minibatch 2500/2500, validation error 17.080000 %
momentum:  0.902662098408
learning rate:  0.029399142974
epoch 82, minibatch 2500/2500, validation error 16.810000 %
momentum:  0.904608845711
learning rate:  0.0289581558294
epoch 83, minibatch 2500/2500, validation error 16.900000 %
momentum:  0.906516671181
learning rate:  0.0285237834919
epoch 84, minibatch 2500/2500, validation error 16.630000 %
momentum:  0.908386349678
learning rate:  0.0280959267395
epoch 85, minibatch 2500/2500, validation error 16.700000 %
momentum:  0.910218596458
learning rate:  0.0276744878384
epoch 86, minibatch 2500/2500, validation error 17.000000 %
momentum:  0.912014245987
learning rate:  0.0272593705209
epoch 87, minibatch 2500/2500, validation error 16.990000 %
momentum:  0.913773953915
learning rate:  0.026850479963
epoch 88, minibatch 2500/2500, validation error 16.730000 %
momentum:  0.915498495102
learning rate:  0.0264477227636
epoch 89, minibatch 2500/2500, validation error 16.750000 %
momentum:  0.9171885252
learning rate:  0.0260510069221
epoch 90, minibatch 2500/2500, validation error 17.020000 %
momentum:  0.918844759464
learning rate:  0.0256602418183
epoch 91, minibatch 2500/2500, validation error 16.900000 %
momentum:  0.920467853546
learning rate:  0.025275338191
epoch 92, minibatch 2500/2500, validation error 16.790000 %
momentum:  0.922058522701
learning rate:  0.0248962081182
epoch 93, minibatch 2500/2500, validation error 16.840000 %
momentum:  0.923617362976
learning rate:  0.0245227649964
epoch 94, minibatch 2500/2500, validation error 16.650000 %
momentum:  0.925145030022
learning rate:  0.0241549235215
epoch 95, minibatch 2500/2500, validation error 16.980000 %
momentum:  0.926642119884
learning rate:  0.0237925996686
epoch 96, minibatch 2500/2500, validation error 16.910000 %
momentum:  0.928109288216
learning rate:  0.0234357106736
epoch 97, minibatch 2500/2500, validation error 17.230000 %
momentum:  0.929547131062
learning rate:  0.0230841750135
epoch 98, minibatch 2500/2500, validation error 17.260000 %
momentum:  0.930956184864
learning rate:  0.0227379123883
epoch 99, minibatch 2500/2500, validation error 16.870000 %
momentum:  0.93233704567
learning rate:  0.0223968437025
epoch 100, minibatch 2500/2500, validation error 16.960000 %
Optimization complete. Best validation score of 16.630000 % obtained at iteration 175000, with test performance 17.940000 %
train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 31.850000 %
     epoch 1, minibatch 2500/2500, test error of best model 33.990000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 21.420000 %
     epoch 2, minibatch 2500/2500, test error of best model 22.830000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 17.600000 %
     epoch 3, minibatch 2500/2500, test error of best model 19.000000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 15.080000 %
     epoch 4, minibatch 2500/2500, test error of best model 16.310000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatch 2500/2500, validation error 12.890000 %
     epoch 5, minibatch 2500/2500, test error of best model 13.790000 %
momentum:  0.548039615154
learning rate:  0.0927216502366
epoch 6, minibatch 2500/2500, validation error 9.980000 %
     epoch 6, minibatch 2500/2500, test error of best model 10.790000 %
momentum:  0.557078838348
learning rate:  0.091330825483
epoch 7, minibatch 2500/2500, validation error 10.840000 %
momentum:  0.565937280655
learning rate:  0.0899608631008
epoch 8, minibatch 2500/2500, validation error 8.880000 %
     epoch 8, minibatch 2500/2500, test error of best model 9.350000 %
momentum:  0.574618518353
learning rate:  0.0886114501543
epoch 9, minibatch 2500/2500, validation error 8.160000 %
     epoch 9, minibatch 2500/2500, test error of best model 8.460000 %
momentum:  0.58312612772
learning rate:  0.0872822784019
epoch 10, minibatch 2500/2500, validation error 7.400000 %
     epoch 10, minibatch 2500/2500, test error of best model 7.710000 %
momentum:  0.591463625431
learning rate:  0.0859730442259
epoch 11, minibatch 2500/2500, validation error 8.030000 %
momentum:  0.599634349346
learning rate:  0.0846834485625
epoch 12, minibatch 2500/2500, validation error 6.890000 %
     epoch 12, minibatch 2500/2500, test error of best model 7.070000 %
momentum:  0.607641637325
learning rate:  0.0834131968341
epoch 13, minibatch 2500/2500, validation error 6.960000 %
momentum:  0.615488827229
learning rate:  0.0821619988816
epoch 14, minibatch 2500/2500, validation error 7.150000 %
momentum:  0.623179078102
learning rate:  0.0809295688984
epoch 15, minibatch 2500/2500, validation error 7.250000 %
momentum:  0.630715489388
learning rate:  0.0797156253649
epoch 16, minibatch 2500/2500, validation error 7.070000 %
momentum:  0.638101160526
learning rate:  0.0785198909844
epoch 17, minibatch 2500/2500, validation error 5.970000 %
     epoch 17, minibatch 2500/2500, test error of best model 6.180000 %
momentum:  0.645339131355
learning rate:  0.0773420926196
epoch 18, minibatch 2500/2500, validation error 5.830000 %
     epoch 18, minibatch 2500/2500, test error of best model 5.790000 %
momentum:  0.652432322502
learning rate:  0.0761819612303
epoch 19, minibatch 2500/2500, validation error 5.650000 %
     epoch 19, minibatch 2500/2500, test error of best model 5.680000 %
momentum:  0.659383654594
learning rate:  0.0750392318119
epoch 20, minibatch 2500/2500, validation error 6.800000 %
momentum:  0.666195988655
learning rate:  0.0739136433347
epoch 21, minibatch 2500/2500, validation error 5.390000 %
     epoch 21, minibatch 2500/2500, test error of best model 5.580000 %
momentum:  0.672872066498
learning rate:  0.0728049386847
epoch 22, minibatch 2500/2500, validation error 4.960000 %
     epoch 22, minibatch 2500/2500, test error of best model 5.210000 %
momentum:  0.679414629936
learning rate:  0.0717128646044
epoch 23, minibatch 2500/2500, validation error 5.570000 %
momentum:  0.685826361179
learning rate:  0.0706371716354
epoch 24, minibatch 2500/2500, validation error 4.760000 %
     epoch 24, minibatch 2500/2500, test error of best model 4.910000 %
momentum:  0.692109823227
learning rate:  0.0695776140608
epoch 25, minibatch 2500/2500, validation error 5.040000 %
momentum:  0.698267638683
learning rate:  0.0685339498499
epoch 26, minibatch 2500/2500, validation error 5.150000 %
momentum:  0.704302310944
learning rate:  0.0675059406022
epoch 27, minibatch 2500/2500, validation error 5.410000 %
momentum:  0.710216283798
learning rate:  0.0664933514931
epoch 28, minibatch 2500/2500, validation error 5.220000 %
momentum:  0.716011941433
learning rate:  0.0654959512207
epoch 29, minibatch 2500/2500, validation error 4.820000 %
momentum:  0.721691727638
learning rate:  0.0645135119524
epoch 30, minibatch 2500/2500, validation error 4.650000 %
     epoch 30, minibatch 2500/2500, test error of best model 4.890000 %
momentum:  0.727257907391
learning rate:  0.0635458092731
epoch 31, minibatch 2500/2500, validation error 4.340000 %
     epoch 31, minibatch 2500/2500, test error of best model 4.610000 %
momentum:  0.732712745667
learning rate:  0.062592622134
epoch 32, minibatch 2500/2500, validation error 5.120000 %
momentum:  0.738058507442
learning rate:  0.061653732802
epoch 33, minibatch 2500/2500, validation error 4.880000 %
momentum:  0.743297338486
learning rate:  0.06072892681
epoch 34, minibatch 2500/2500, validation error 4.310000 %
     epoch 34, minibatch 2500/2500, test error of best model 4.580000 %
momentum:  0.748431384563
learning rate:  0.0598179929078
epoch 35, minibatch 2500/2500, validation error 5.250000 %
momentum:  0.753462731838
learning rate:  0.0589207230142
epoch 36, minibatch 2500/2500, validation error 4.720000 %
momentum:  0.758393466473
learning rate:  0.058036912169
epoch 37, minibatch 2500/2500, validation error 4.270000 %
     epoch 37, minibatch 2500/2500, test error of best model 4.700000 %
momentum:  0.763225615025
learning rate:  0.0571663584865
epoch 38, minibatch 2500/2500, validation error 4.680000 %
momentum:  0.767961084843
learning rate:  0.0563088631092
epoch 39, minibatch 2500/2500, validation error 4.560000 %
momentum:  0.77260184288
learning rate:  0.0554642301625
epoch 40, minibatch 2500/2500, validation error 4.270000 %
momentum:  0.777149796486
learning rate:  0.0546322667101
epoch 41, minibatch 2500/2500, validation error 4.210000 %
     epoch 41, minibatch 2500/2500, test error of best model 4.390000 %
momentum:  0.781606793404
learning rate:  0.0538127827095
epoch 42, minibatch 2500/2500, validation error 4.160000 %
     epoch 42, minibatch 2500/2500, test error of best model 4.170000 %
momentum:  0.785974681377
learning rate:  0.0530055909688
epoch 43, minibatch 2500/2500, validation error 3.860000 %
     epoch 43, minibatch 2500/2500, test error of best model 4.070000 %
momentum:  0.790255188942
learning rate:  0.0522105071043
epoch 44, minibatch 2500/2500, validation error 4.520000 %
momentum:  0.794450104237
learning rate:  0.0514273494977
epoch 45, minibatch 2500/2500, validation error 4.120000 %
momentum:  0.798561096191
learning rate:  0.0506559392553
epoch 46, minibatch 2500/2500, validation error 4.920000 %
momentum:  0.802589893341
learning rate:  0.0498961001664
epoch 47, minibatch 2500/2500, validation error 4.500000 %
momentum:  0.806538105011
learning rate:  0.0491476586639
epoch 48, minibatch 2500/2500, validation error 4.140000 %
momentum:  0.810407340527
learning rate:  0.048410443784
epoch 49, minibatch 2500/2500, validation error 4.110000 %
momentum:  0.814199209213
learning rate:  0.0476842871272
epoch 50, minibatch 2500/2500, validation error 4.030000 %
momentum:  0.817915201187
learning rate:  0.0469690228203
epoch 51, minibatch 2500/2500, validation error 4.160000 %
momentum:  0.821556925774
learning rate:  0.046264487478
epoch 52, minibatch 2500/2500, validation error 3.980000 %
momentum:  0.825125813484
learning rate:  0.0455705201658
epoch 53, minibatch 2500/2500, validation error 4.140000 %
momentum:  0.82862329483
learning rate:  0.0448869623633
epoch 54, minibatch 2500/2500, validation error 3.890000 %
momentum:  0.832050800323
learning rate:  0.0442136579279
epoch 55, minibatch 2500/2500, validation error 4.000000 %
momentum:  0.835409760475
learning rate:  0.043550453059
epoch 56, minibatch 2500/2500, validation error 4.240000 %
momentum:  0.838701546192
learning rate:  0.0428971962631
epoch 57, minibatch 2500/2500, validation error 3.860000 %
     epoch 57, minibatch 2500/2500, test error of best model 4.000000 %
momentum:  0.841927528381
learning rate:  0.0422537383191
epoch 58, minibatch 2500/2500, validation error 4.140000 %
momentum:  0.84508895874
learning rate:  0.0416199322444
epoch 59, minibatch 2500/2500, validation error 3.840000 %
     epoch 59, minibatch 2500/2500, test error of best model 4.050000 %
momentum:  0.848187208176
learning rate:  0.0409956332607
epoch 60, minibatch 2500/2500, validation error 4.190000 %
momentum:  0.851223468781
learning rate:  0.0403806987618
epoch 61, minibatch 2500/2500, validation error 3.810000 %
     epoch 61, minibatch 2500/2500, test error of best model 4.070000 %
momentum:  0.854198992252
learning rate:  0.0397749882803
epoch 62, minibatch 2500/2500, validation error 3.810000 %
momentum:  0.857115030289
learning rate:  0.0391783634561
epoch 63, minibatch 2500/2500, validation error 4.430000 %
momentum:  0.859972715378
learning rate:  0.0385906880043
epoch 64, minibatch 2500/2500, validation error 4.090000 %
momentum:  0.862773239613
learning rate:  0.0380118276842
epoch 65, minibatch 2500/2500, validation error 3.840000 %
momentum:  0.865517795086
learning rate:  0.037441650269
epoch 66, minibatch 2500/2500, validation error 4.160000 %
momentum:  0.868207454681
learning rate:  0.0368800255149
epoch 67, minibatch 2500/2500, validation error 4.330000 %
momentum:  0.870843291283
learning rate:  0.0363268251322
epoch 68, minibatch 2500/2500, validation error 4.240000 %
momentum:  0.873426437378
learning rate:  0.0357819227552
epoch 69, minibatch 2500/2500, validation error 4.570000 %
momentum:  0.875957906246
learning rate:  0.0352451939139
epoch 70, minibatch 2500/2500, validation error 4.110000 %
momentum:  0.878438770771
learning rate:  0.0347165160052
epoch 71, minibatch 2500/2500, validation error 4.180000 %
momentum:  0.880869984627
learning rate:  0.0341957682651
epoch 72, minibatch 2500/2500, validation error 3.920000 %
momentum:  0.883252561092
learning rate:  0.0336828317411
epoch 73, minibatch 2500/2500, validation error 4.510000 %
momentum:  0.885587513447
learning rate:  0.033177589265
epoch 74, minibatch 2500/2500, validation error 4.530000 %
momentum:  0.88787573576
learning rate:  0.032679925426
epoch 75, minibatch 2500/2500, validation error 4.310000 %
momentum:  0.89011824131
learning rate:  0.0321897265447
epoch 76, minibatch 2500/2500, validation error 4.280000 %
momentum:  0.892315864563
learning rate:  0.0317068806465
epoch 77, minibatch 2500/2500, validation error 4.340000 %
momentum:  0.894469559193
learning rate:  0.0312312774368
epoch 78, minibatch 2500/2500, validation error 4.450000 %
momentum:  0.896580159664
learning rate:  0.0307628082752
epoch 79, minibatch 2500/2500, validation error 4.330000 %
momentum:  0.898648560047
learning rate:  0.0303013661511
epoch 80, minibatch 2500/2500, validation error 4.290000 %
momentum:  0.900675594807
learning rate:  0.0298468456588
epoch 81, minibatch 2500/2500, validation error 4.280000 %
momentum:  0.902662098408
learning rate:  0.029399142974
epoch 82, minibatch 2500/2500, validation error 3.810000 %
momentum:  0.904608845711
learning rate:  0.0289581558294
epoch 83, minibatch 2500/2500, validation error 3.810000 %
momentum:  0.906516671181
learning rate:  0.0285237834919
epoch 84, minibatch 2500/2500, validation error 4.440000 %
momentum:  0.908386349678
learning rate:  0.0280959267395
epoch 85, minibatch 2500/2500, validation error 4.170000 %
momentum:  0.910218596458
learning rate:  0.0276744878384
epoch 86, minibatch 2500/2500, validation error 3.880000 %
momentum:  0.912014245987
learning rate:  0.0272593705209
epoch 87, minibatch 2500/2500, validation error 3.940000 %
momentum:  0.913773953915
learning rate:  0.026850479963
epoch 88, minibatch 2500/2500, validation error 4.110000 %
momentum:  0.915498495102
learning rate:  0.0264477227636
epoch 89, minibatch 2500/2500, validation error 4.010000 %
momentum:  0.9171885252
learning rate:  0.0260510069221
epoch 90, minibatch 2500/2500, validation error 4.190000 %
momentum:  0.918844759464
learning rate:  0.0256602418183
epoch 91, minibatch 2500/2500, validation error 3.900000 %
momentum:  0.920467853546
learning rate:  0.025275338191
epoch 92, minibatch 2500/2500, validation error 3.910000 %
momentum:  0.922058522701
learning rate:  0.0248962081182
epoch 93, minibatch 2500/2500, validation error 4.000000 %
momentum:  0.923617362976
learning rate:  0.0245227649964
epoch 94, minibatch 2500/2500, validation error 4.490000 %
momentum:  0.925145030022
learning rate:  0.0241549235215
epoch 95, minibatch 2500/2500, validation error 3.940000 %
momentum:  0.926642119884
learning rate:  0.0237925996686
epoch 96, minibatch 2500/2500, validation error 4.200000 %
momentum:  0.928109288216
learning rate:  0.0234357106736
epoch 97, minibatch 2500/2500, validation error 4.810000 %
momentum:  0.929547131062
learning rate:  0.0230841750135
epoch 98, minibatch 2500/2500, validation error 4.250000 %
momentum:  0.930956184864
learning rate:  0.0227379123883
epoch 99, minibatch 2500/2500, validation error 4.320000 %
momentum:  0.93233704567
learning rate:  0.0223968437025
epoch 100, minibatch 2500/2500, validation error 4.000000 %
Optimization complete. Best validation score of 3.810000 % obtained at iteration 152500, with test performance 4.070000 %
train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 9.510000 %
     epoch 1, minibatch 2500/2500, test error of best model 10.480000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 6.580000 %
     epoch 2, minibatch 2500/2500, test error of best model 7.530000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 5.010000 %
     epoch 3, minibatch 2500/2500, test error of best model 5.770000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 4.100000 %
     epoch 4, minibatch 2500/2500, test error of best model 4.690000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatch 2500/2500, validation error 3.810000 %
     epoch 5, minibatch 2500/2500, test error of best model 4.150000 %
momentum:  0.548039615154
learning rate:  0.0927216502366
epoch 6, minibatch 2500/2500, validation error 3.490000 %
     epoch 6, minibatch 2500/2500, test error of best model 3.850000 %
momentum:  0.557078838348
learning rate:  0.091330825483
epoch 7, minibatch 2500/2500, validation error 3.200000 %
     epoch 7, minibatch 2500/2500, test error of best model 3.550000 %
momentum:  0.565937280655
learning rate:  0.0899608631008
epoch 8, minibatch 2500/2500, validation error 3.180000 %
     epoch 8, minibatch 2500/2500, test error of best model 3.330000 %
momentum:  0.574618518353
learning rate:  0.0886114501543
epoch 9, minibatch 2500/2500, validation error 3.010000 %
     epoch 9, minibatch 2500/2500, test error of best model 3.250000 %
momentum:  0.58312612772
learning rate:  0.0872822784019
epoch 10, minibatch 2500/2500, validation error 3.030000 %
momentum:  0.591463625431
learning rate:  0.0859730442259
epoch 11, minibatch 2500/2500, validation error 2.860000 %
     epoch 11, minibatch 2500/2500, test error of best model 3.120000 %
momentum:  0.599634349346
learning rate:  0.0846834485625
epoch 12, minibatch 2500/2500, validation error 2.820000 %
     epoch 12, minibatch 2500/2500, test error of best model 3.150000 %
momentum:  0.607641637325
learning rate:  0.0834131968341
epoch 13, minibatch 2500/2500, validation error 2.660000 %
     epoch 13, minibatch 2500/2500, test error of best model 2.940000 %
momentum:  0.615488827229
learning rate:  0.0821619988816
epoch 14, minibatch 2500/2500, validation error 2.700000 %
momentum:  0.623179078102
learning rate:  0.0809295688984
epoch 15, minibatch 2500/2500, validation error 2.790000 %
momentum:  0.630715489388
learning rate:  0.0797156253649
epoch 16, minibatch 2500/2500, validation error 2.630000 %
     epoch 16, minibatch 2500/2500, test error of best model 2.960000 %
momentum:  0.638101160526
learning rate:  0.0785198909844
epoch 17, minibatch 2500/2500, validation error 2.690000 %
momentum:  0.645339131355
learning rate:  0.0773420926196
epoch 18, minibatch 2500/2500, validation error 2.680000 %
momentum:  0.652432322502
learning rate:  0.0761819612303
epoch 19, minibatch 2500/2500, validation error 2.520000 %
     epoch 19, minibatch 2500/2500, test error of best model 2.760000 %
momentum:  0.659383654594
learning rate:  0.0750392318119
epoch 20, minibatch 2500/2500, validation error 2.510000 %
     epoch 20, minibatch 2500/2500, test error of best model 2.770000 %
momentum:  0.666195988655
learning rate:  0.0739136433347
epoch 21, minibatch 2500/2500, validation error 2.560000 %
momentum:  0.672872066498
learning rate:  0.0728049386847
epoch 22, minibatch 2500/2500, validation error 2.490000 %
     epoch 22, minibatch 2500/2500, test error of best model 2.810000 %
momentum:  0.679414629936
learning rate:  0.0717128646044
epoch 23, minibatch 2500/2500, validation error 2.650000 %
momentum:  0.685826361179
learning rate:  0.0706371716354
epoch 24, minibatch 2500/2500, validation error 2.590000 %
momentum:  0.692109823227
learning rate:  0.0695776140608
epoch 25, minibatch 2500/2500, validation error 2.440000 %
     epoch 25, minibatch 2500/2500, test error of best model 2.550000 %
momentum:  0.698267638683
learning rate:  0.0685339498499
epoch 26, minibatch 2500/2500, validation error 2.430000 %
     epoch 26, minibatch 2500/2500, test error of best model 2.550000 %
momentum:  0.704302310944
learning rate:  0.0675059406022
epoch 27, minibatch 2500/2500, validation error 2.480000 %
momentum:  0.710216283798
learning rate:  0.0664933514931
epoch 28, minibatch 2500/2500, validation error 2.470000 %
momentum:  0.716011941433
learning rate:  0.0654959512207
epoch 29, minibatch 2500/2500, validation error 2.420000 %
     epoch 29, minibatch 2500/2500, test error of best model 2.650000 %
momentum:  0.721691727638
learning rate:  0.0645135119524
epoch 30, minibatch 2500/2500, validation error 2.630000 %
momentum:  0.727257907391
learning rate:  0.0635458092731
epoch 31, minibatch 2500/2500, validation error 2.520000 %
momentum:  0.732712745667
learning rate:  0.062592622134
epoch 32, minibatch 2500/2500, validation error 2.500000 %
momentum:  0.738058507442
learning rate:  0.061653732802
epoch 33, minibatch 2500/2500, validation error 2.460000 %
momentum:  0.743297338486
learning rate:  0.06072892681
epoch 34, minibatch 2500/2500, validation error 2.410000 %
     epoch 34, minibatch 2500/2500, test error of best model 2.580000 %
momentum:  0.748431384563
learning rate:  0.0598179929078
epoch 35, minibatch 2500/2500, validation error 2.590000 %
momentum:  0.753462731838
learning rate:  0.0589207230142
epoch 36, minibatch 2500/2500, validation error 2.440000 %
momentum:  0.758393466473
learning rate:  0.058036912169
epoch 37, minibatch 2500/2500, validation error 2.560000 %
momentum:  0.763225615025
learning rate:  0.0571663584865
epoch 38, minibatch 2500/2500, validation error 2.690000 %
momentum:  0.767961084843
learning rate:  0.0563088631092
epoch 39, minibatch 2500/2500, validation error 2.380000 %
     epoch 39, minibatch 2500/2500, test error of best model 2.560000 %
momentum:  0.77260184288
learning rate:  0.0554642301625
epoch 40, minibatch 2500/2500, validation error 2.460000 %
momentum:  0.777149796486
learning rate:  0.0546322667101
epoch 41, minibatch 2500/2500, validation error 2.420000 %
momentum:  0.781606793404
learning rate:  0.0538127827095
epoch 42, minibatch 2500/2500, validation error 2.530000 %
momentum:  0.785974681377
learning rate:  0.0530055909688
epoch 43, minibatch 2500/2500, validation error 2.600000 %
momentum:  0.790255188942
learning rate:  0.0522105071043
epoch 44, minibatch 2500/2500, validation error 2.560000 %
momentum:  0.794450104237
learning rate:  0.0514273494977
epoch 45, minibatch 2500/2500, validation error 2.450000 %
momentum:  0.798561096191
learning rate:  0.0506559392553
epoch 46, minibatch 2500/2500, validation error 2.360000 %
     epoch 46, minibatch 2500/2500, test error of best model 2.440000 %
momentum:  0.802589893341
learning rate:  0.0498961001664
epoch 47, minibatch 2500/2500, validation error 2.470000 %
momentum:  0.806538105011
learning rate:  0.0491476586639
epoch 48, minibatch 2500/2500, validation error 2.480000 %
momentum:  0.810407340527
learning rate:  0.048410443784
epoch 49, minibatch 2500/2500, validation error 2.400000 %
momentum:  0.814199209213
learning rate:  0.0476842871272
epoch 50, minibatch 2500/2500, validation error 2.620000 %
momentum:  0.817915201187
learning rate:  0.0469690228203
epoch 51, minibatch 2500/2500, validation error 2.210000 %
     epoch 51, minibatch 2500/2500, test error of best model 2.510000 %
momentum:  0.821556925774
learning rate:  0.046264487478
epoch 52, minibatch 2500/2500, validation error 2.300000 %
momentum:  0.825125813484
learning rate:  0.0455705201658
epoch 53, minibatch 2500/2500, validation error 2.460000 %
momentum:  0.82862329483
learning rate:  0.0448869623633
epoch 54, minibatch 2500/2500, validation error 2.690000 %
momentum:  0.832050800323
learning rate:  0.0442136579279
epoch 55, minibatch 2500/2500, validation error 2.510000 %
momentum:  0.835409760475
learning rate:  0.043550453059
epoch 56, minibatch 2500/2500, validation error 2.210000 %
momentum:  0.838701546192
learning rate:  0.0428971962631
epoch 57, minibatch 2500/2500, validation error 2.590000 %
momentum:  0.841927528381
learning rate:  0.0422537383191
epoch 58, minibatch 2500/2500, validation error 2.390000 %
momentum:  0.84508895874
learning rate:  0.0416199322444
epoch 59, minibatch 2500/2500, validation error 2.490000 %
momentum:  0.848187208176
learning rate:  0.0409956332607
epoch 60, minibatch 2500/2500, validation error 2.710000 %
momentum:  0.851223468781
learning rate:  0.0403806987618
epoch 61, minibatch 2500/2500, validation error 2.470000 %
momentum:  0.854198992252
learning rate:  0.0397749882803
epoch 62, minibatch 2500/2500, validation error 2.440000 %
momentum:  0.857115030289
learning rate:  0.0391783634561
epoch 63, minibatch 2500/2500, validation error 2.500000 %
momentum:  0.859972715378
learning rate:  0.0385906880043
epoch 64, minibatch 2500/2500, validation error 2.390000 %
momentum:  0.862773239613
learning rate:  0.0380118276842
epoch 65, minibatch 2500/2500, validation error 2.430000 %
momentum:  0.865517795086
learning rate:  0.037441650269
epoch 66, minibatch 2500/2500, validation error 2.550000 %
momentum:  0.868207454681
learning rate:  0.0368800255149
epoch 67, minibatch 2500/2500, validation error 2.650000 %
momentum:  0.870843291283
learning rate:  0.0363268251322
epoch 68, minibatch 2500/2500, validation error 2.400000 %
momentum:  0.873426437378
learning rate:  0.0357819227552
epoch 69, minibatch 2500/2500, validation error 2.410000 %
momentum:  0.875957906246
learning rate:  0.0352451939139
epoch 70, minibatch 2500/2500, validation error 2.500000 %
momentum:  0.878438770771
learning rate:  0.0347165160052
epoch 71, minibatch 2500/2500, validation error 2.470000 %
momentum:  0.880869984627
learning rate:  0.0341957682651
epoch 72, minibatch 2500/2500, validation error 2.360000 %
momentum:  0.883252561092
learning rate:  0.0336828317411
epoch 73, minibatch 2500/2500, validation error 2.250000 %
momentum:  0.885587513447
learning rate:  0.033177589265
epoch 74, minibatch 2500/2500, validation error 2.420000 %
momentum:  0.88787573576
learning rate:  0.032679925426
epoch 75, minibatch 2500/2500, validation error 2.410000 %
momentum:  0.89011824131
learning rate:  0.0321897265447
epoch 76, minibatch 2500/2500, validation error 2.430000 %
momentum:  0.892315864563
learning rate:  0.0317068806465
epoch 77, minibatch 2500/2500, validation error 2.540000 %
momentum:  0.894469559193
learning rate:  0.0312312774368
epoch 78, minibatch 2500/2500, validation error 2.490000 %
momentum:  0.896580159664
learning rate:  0.0307628082752
epoch 79, minibatch 2500/2500, validation error 2.450000 %
momentum:  0.898648560047
learning rate:  0.0303013661511
epoch 80, minibatch 2500/2500, validation error 2.360000 %
momentum:  0.900675594807
learning rate:  0.0298468456588
epoch 81, minibatch 2500/2500, validation error 2.520000 %
momentum:  0.902662098408
learning rate:  0.029399142974
epoch 82, minibatch 2500/2500, validation error 2.450000 %
momentum:  0.904608845711
learning rate:  0.0289581558294
epoch 83, minibatch 2500/2500, validation error 2.410000 %
momentum:  0.906516671181
learning rate:  0.0285237834919
epoch 84, minibatch 2500/2500, validation error 2.550000 %
momentum:  0.908386349678
learning rate:  0.0280959267395
epoch 85, minibatch 2500/2500, validation error 2.570000 %
momentum:  0.910218596458
learning rate:  0.0276744878384
epoch 86, minibatch 2500/2500, validation error 2.440000 %
momentum:  0.912014245987
learning rate:  0.0272593705209
epoch 87, minibatch 2500/2500, validation error 2.460000 %
momentum:  0.913773953915
learning rate:  0.026850479963
epoch 88, minibatch 2500/2500, validation error 2.560000 %
momentum:  0.915498495102
learning rate:  0.0264477227636
epoch 89, minibatch 2500/2500, validation error 2.550000 %
momentum:  0.9171885252
learning rate:  0.0260510069221
epoch 90, minibatch 2500/2500, validation error 2.500000 %
momentum:  0.918844759464
learning rate:  0.0256602418183
epoch 91, minibatch 2500/2500, validation error 2.470000 %
momentum:  0.920467853546
learning rate:  0.025275338191
epoch 92, minibatch 2500/2500, validation error 2.420000 %
momentum:  0.922058522701
learning rate:  0.0248962081182
epoch 93, minibatch 2500/2500, validation error 2.680000 %
momentum:  0.923617362976
learning rate:  0.0245227649964
epoch 94, minibatch 2500/2500, validation error 2.830000 %
momentum:  0.925145030022
learning rate:  0.0241549235215
epoch 95, minibatch 2500/2500, validation error 2.610000 %
momentum:  0.926642119884
learning rate:  0.0237925996686
epoch 96, minibatch 2500/2500, validation error 2.840000 %
momentum:  0.928109288216
learning rate:  0.0234357106736
epoch 97, minibatch 2500/2500, validation error 2.540000 %
momentum:  0.929547131062
learning rate:  0.0230841750135
epoch 98, minibatch 2500/2500, validation error 2.760000 %
momentum:  0.930956184864
learning rate:  0.0227379123883
epoch 99, minibatch 2500/2500, validation error 2.620000 %
momentum:  0.93233704567
learning rate:  0.0223968437025
epoch 100, minibatch 2500/2500, validation error 2.570000 %
Optimization complete. Best validation score of 2.210000 % obtained at iteration 127500, with test performance 2.510000 %
train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 7.230000 %
     epoch 1, minibatch 2500/2500, test error of best model 7.880000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 4.560000 %
     epoch 2, minibatch 2500/2500, test error of best model 5.300000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 3.690000 %
     epoch 3, minibatch 2500/2500, test error of best model 4.240000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 3.390000 %
     epoch 4, minibatch 2500/2500, test error of best model 3.970000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatch 2500/2500, validation error 2.940000 %
     epoch 5, minibatch 2500/2500, test error of best model 3.170000 %
momentum:  0.548039615154
learning rate:  0.0927216502366
epoch 6, minibatch 2500/2500, validation error 2.810000 %
     epoch 6, minibatch 2500/2500, test error of best model 3.140000 %
momentum:  0.557078838348
learning rate:  0.091330825483
epoch 7, minibatch 2500/2500, validation error 2.650000 %
     epoch 7, minibatch 2500/2500, test error of best model 2.820000 %
momentum:  0.565937280655
learning rate:  0.0899608631008
epoch 8, minibatch 2500/2500, validation error 2.790000 %
momentum:  0.574618518353
learning rate:  0.0886114501543
epoch 9, minibatch 2500/2500, validation error 2.490000 %
     epoch 9, minibatch 2500/2500, test error of best model 2.510000 %
momentum:  0.58312612772
learning rate:  0.0872822784019
epoch 10, minibatch 2500/2500, validation error 2.510000 %
momentum:  0.591463625431
learning rate:  0.0859730442259
epoch 11, minibatch 2500/2500, validation error 2.300000 %
     epoch 11, minibatch 2500/2500, test error of best model 2.380000 %
momentum:  0.599634349346
learning rate:  0.0846834485625
epoch 12, minibatch 2500/2500, validation error 2.360000 %
momentum:  0.607641637325
learning rate:  0.0834131968341
epoch 13, minibatch 2500/2500, validation error 2.350000 %
momentum:  0.615488827229
learning rate:  0.0821619988816
epoch 14, minibatch 2500/2500, validation error 2.460000 %
momentum:  0.623179078102
learning rate:  0.0809295688984
epoch 15, minibatch 2500/2500, validation error 2.260000 %
     epoch 15, minibatch 2500/2500, test error of best model 2.330000 %
momentum:  0.630715489388
learning rate:  0.0797156253649
epoch 16, minibatch 2500/2500, validation error 2.380000 %
momentum:  0.638101160526
learning rate:  0.0785198909844
epoch 17, minibatch 2500/2500, validation error 2.220000 %
     epoch 17, minibatch 2500/2500, test error of best model 2.340000 %
momentum:  0.645339131355
learning rate:  0.0773420926196
epoch 18, minibatch 2500/2500, validation error 2.160000 %
     epoch 18, minibatch 2500/2500, test error of best model 2.380000 %
momentum:  0.652432322502
learning rate:  0.0761819612303
epoch 19, minibatch 2500/2500, validation error 2.210000 %
momentum:  0.659383654594
learning rate:  0.0750392318119
epoch 20, minibatch 2500/2500, validation error 2.050000 %
     epoch 20, minibatch 2500/2500, test error of best model 2.280000 %
momentum:  0.666195988655
learning rate:  0.0739136433347
epoch 21, minibatch 2500/2500, validation error 2.170000 %
momentum:  0.672872066498
learning rate:  0.0728049386847
epoch 22, minibatch 2500/2500, validation error 2.160000 %
momentum:  0.679414629936
learning rate:  0.0717128646044
epoch 23, minibatch 2500/2500, validation error 2.030000 %
     epoch 23, minibatch 2500/2500, test error of best model 2.170000 %
momentum:  0.685826361179
learning rate:  0.0706371716354
epoch 24, minibatch 2500/2500, validation error 2.140000 %
momentum:  0.692109823227
learning rate:  0.0695776140608
epoch 25, minibatch 2500/2500, validation error 2.290000 %
momentum:  0.698267638683
learning rate:  0.0685339498499
epoch 26, minibatch 2500/2500, validation error 2.130000 %
momentum:  0.704302310944
learning rate:  0.0675059406022
epoch 27, minibatch 2500/2500, validation error 2.170000 %
momentum:  0.710216283798
learning rate:  0.0664933514931
epoch 28, minibatch 2500/2500, validation error 2.340000 %
momentum:  0.716011941433
learning rate:  0.0654959512207
epoch 29, minibatch 2500/2500, validation error 2.140000 %
momentum:  0.721691727638
learning rate:  0.0645135119524
epoch 30, minibatch 2500/2500, validation error 2.230000 %
momentum:  0.727257907391
learning rate:  0.0635458092731
epoch 31, minibatch 2500/2500, validation error 2.080000 %
momentum:  0.732712745667
learning rate:  0.062592622134
epoch 32, minibatch 2500/2500, validation error 1.980000 %
     epoch 32, minibatch 2500/2500, test error of best model 2.140000 %
momentum:  0.738058507442
learning rate:  0.061653732802
epoch 33, minibatch 2500/2500, validation error 2.040000 %
momentum:  0.743297338486
learning rate:  0.06072892681
epoch 34, minibatch 2500/2500, validation error 2.040000 %
momentum:  0.748431384563
learning rate:  0.0598179929078
epoch 35, minibatch 2500/2500, validation error 2.040000 %
momentum:  0.753462731838
learning rate:  0.0589207230142
epoch 36, minibatch 2500/2500, validation error 2.030000 %
momentum:  0.758393466473
learning rate:  0.058036912169
epoch 37, minibatch 2500/2500, validation error 2.120000 %
momentum:  0.763225615025
learning rate:  0.0571663584865
epoch 38, minibatch 2500/2500, validation error 2.120000 %
momentum:  0.767961084843
learning rate:  0.0563088631092
epoch 39, minibatch 2500/2500, validation error 2.090000 %
momentum:  0.77260184288
learning rate:  0.0554642301625
epoch 40, minibatch 2500/2500, validation error 2.030000 %
momentum:  0.777149796486
learning rate:  0.0546322667101
epoch 41, minibatch 2500/2500, validation error 2.150000 %
momentum:  0.781606793404
learning rate:  0.0538127827095
epoch 42, minibatch 2500/2500, validation error 1.990000 %
momentum:  0.785974681377
learning rate:  0.0530055909688
epoch 43, minibatch 2500/2500, validation error 2.110000 %
momentum:  0.790255188942
learning rate:  0.0522105071043
epoch 44, minibatch 2500/2500, validation error 2.130000 %
momentum:  0.794450104237
learning rate:  0.0514273494977
epoch 45, minibatch 2500/2500, validation error 2.050000 %
momentum:  0.798561096191
learning rate:  0.0506559392553
epoch 46, minibatch 2500/2500, validation error 2.010000 %
momentum:  0.802589893341
learning rate:  0.0498961001664
epoch 47, minibatch 2500/2500, validation error 1.970000 %
     epoch 47, minibatch 2500/2500, test error of best model 2.150000 %
momentum:  0.806538105011
learning rate:  0.0491476586639
epoch 48, minibatch 2500/2500, validation error 1.880000 %
     epoch 48, minibatch 2500/2500, test error of best model 2.000000 %
momentum:  0.810407340527
learning rate:  0.048410443784
epoch 49, minibatch 2500/2500, validation error 2.140000 %
momentum:  0.814199209213
learning rate:  0.0476842871272
epoch 50, minibatch 2500/2500, validation error 1.940000 %
momentum:  0.817915201187
learning rate:  0.0469690228203
epoch 51, minibatch 2500/2500, validation error 2.160000 %
momentum:  0.821556925774
learning rate:  0.046264487478
epoch 52, minibatch 2500/2500, validation error 2.130000 %
momentum:  0.825125813484
learning rate:  0.0455705201658
epoch 53, minibatch 2500/2500, validation error 2.100000 %
momentum:  0.82862329483
learning rate:  0.0448869623633
epoch 54, minibatch 2500/2500, validation error 2.060000 %
momentum:  0.832050800323
learning rate:  0.0442136579279
epoch 55, minibatch 2500/2500, validation error 2.080000 %
momentum:  0.835409760475
learning rate:  0.043550453059
epoch 56, minibatch 2500/2500, validation error 2.150000 %
momentum:  0.838701546192
learning rate:  0.0428971962631
epoch 57, minibatch 2500/2500, validation error 2.070000 %
momentum:  0.841927528381
learning rate:  0.0422537383191
epoch 58, minibatch 2500/2500, validation error 2.170000 %
momentum:  0.84508895874
learning rate:  0.0416199322444
epoch 59, minibatch 2500/2500, validation error 2.180000 %
momentum:  0.848187208176
learning rate:  0.0409956332607
epoch 60, minibatch 2500/2500, validation error 2.160000 %
momentum:  0.851223468781
learning rate:  0.0403806987618
epoch 61, minibatch 2500/2500, validation error 1.880000 %
momentum:  0.854198992252
learning rate:  0.0397749882803
epoch 62, minibatch 2500/2500, validation error 2.000000 %
momentum:  0.857115030289
learning rate:  0.0391783634561
epoch 63, minibatch 2500/2500, validation error 1.860000 %
     epoch 63, minibatch 2500/2500, test error of best model 2.190000 %
momentum:  0.859972715378
learning rate:  0.0385906880043
epoch 64, minibatch 2500/2500, validation error 2.110000 %
momentum:  0.862773239613
learning rate:  0.0380118276842
epoch 65, minibatch 2500/2500, validation error 2.070000 %
momentum:  0.865517795086
learning rate:  0.037441650269
epoch 66, minibatch 2500/2500, validation error 1.930000 %
momentum:  0.868207454681
learning rate:  0.0368800255149
epoch 67, minibatch 2500/2500, validation error 2.030000 %
momentum:  0.870843291283
learning rate:  0.0363268251322
epoch 68, minibatch 2500/2500, validation error 1.980000 %
momentum:  0.873426437378
learning rate:  0.0357819227552
epoch 69, minibatch 2500/2500, validation error 2.460000 %
momentum:  0.875957906246
learning rate:  0.0352451939139
epoch 70, minibatch 2500/2500, validation error 1.940000 %
momentum:  0.878438770771
learning rate:  0.0347165160052
epoch 71, minibatch 2500/2500, validation error 2.000000 %
momentum:  0.880869984627
learning rate:  0.0341957682651
epoch 72, minibatch 2500/2500, validation error 2.090000 %
momentum:  0.883252561092
learning rate:  0.0336828317411
epoch 73, minibatch 2500/2500, validation error 1.940000 %
momentum:  0.885587513447
learning rate:  0.033177589265
epoch 74, minibatch 2500/2500, validation error 1.960000 %
momentum:  0.88787573576
learning rate:  0.032679925426
epoch 75, minibatch 2500/2500, validation error 2.110000 %
momentum:  0.89011824131
learning rate:  0.0321897265447
epoch 76, minibatch 2500/2500, validation error 2.360000 %
momentum:  0.892315864563
learning rate:  0.0317068806465
epoch 77, minibatch 2500/2500, validation error 1.950000 %
momentum:  0.894469559193
learning rate:  0.0312312774368
epoch 78, minibatch 2500/2500, validation error 2.120000 %
momentum:  0.896580159664
learning rate:  0.0307628082752
epoch 79, minibatch 2500/2500, validation error 1.980000 %
momentum:  0.898648560047
learning rate:  0.0303013661511
epoch 80, minibatch 2500/2500, validation error 2.130000 %
momentum:  0.900675594807
learning rate:  0.0298468456588
epoch 81, minibatch 2500/2500, validation error 2.010000 %
momentum:  0.902662098408
learning rate:  0.029399142974
epoch 82, minibatch 2500/2500, validation error 1.930000 %
momentum:  0.904608845711
learning rate:  0.0289581558294
epoch 83, minibatch 2500/2500, validation error 2.100000 %
momentum:  0.906516671181
learning rate:  0.0285237834919
epoch 84, minibatch 2500/2500, validation error 1.990000 %
momentum:  0.908386349678
learning rate:  0.0280959267395
epoch 85, minibatch 2500/2500, validation error 2.190000 %
momentum:  0.910218596458
learning rate:  0.0276744878384
epoch 86, minibatch 2500/2500, validation error 2.100000 %
momentum:  0.912014245987
learning rate:  0.0272593705209
epoch 87, minibatch 2500/2500, validation error 1.960000 %
momentum:  0.913773953915
learning rate:  0.026850479963
epoch 88, minibatch 2500/2500, validation error 2.000000 %
momentum:  0.915498495102
learning rate:  0.0264477227636
epoch 89, minibatch 2500/2500, validation error 2.040000 %
momentum:  0.9171885252
learning rate:  0.0260510069221
epoch 90, minibatch 2500/2500, validation error 2.210000 %
momentum:  0.918844759464
learning rate:  0.0256602418183
epoch 91, minibatch 2500/2500, validation error 2.060000 %
momentum:  0.920467853546
learning rate:  0.025275338191
epoch 92, minibatch 2500/2500, validation error 1.990000 %
momentum:  0.922058522701
learning rate:  0.0248962081182
epoch 93, minibatch 2500/2500, validation error 2.100000 %
momentum:  0.923617362976
learning rate:  0.0245227649964
epoch 94, minibatch 2500/2500, validation error 1.910000 %
momentum:  0.925145030022
learning rate:  0.0241549235215
epoch 95, minibatch 2500/2500, validation error 2.090000 %
momentum:  0.926642119884
learning rate:  0.0237925996686
epoch 96, minibatch 2500/2500, validation error 1.980000 %
momentum:  0.928109288216
learning rate:  0.0234357106736
epoch 97, minibatch 2500/2500, validation error 2.080000 %
momentum:  0.929547131062
learning rate:  0.0230841750135
epoch 98, minibatch 2500/2500, validation error 2.050000 %
momentum:  0.930956184864
learning rate:  0.0227379123883
epoch 99, minibatch 2500/2500, validation error 2.210000 %
momentum:  0.93233704567
learning rate:  0.0223968437025
epoch 100, minibatch 2500/2500, validation error 2.050000 %
Optimization complete. Best validation score of 1.860000 % obtained at iteration 157500, with test performance 2.190000 %
train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 6.030000 %
     epoch 1, minibatch 2500/2500, test error of best model 6.760000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 3.950000 %
     epoch 2, minibatch 2500/2500, test error of best model 4.510000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 3.470000 %
     epoch 3, minibatch 2500/2500, test error of best model 3.710000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 3.010000 %
     epoch 4, minibatch 2500/2500, test error of best model 3.580000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatch 2500/2500, validation error 2.670000 %
     epoch 5, minibatch 2500/2500, test error of best model 2.930000 %
momentum:  0.548039615154
learning rate:  0.0927216502366
epoch 6, minibatch 2500/2500, validation error 2.820000 %
momentum:  0.557078838348
learning rate:  0.091330825483
epoch 7, minibatch 2500/2500, validation error 2.490000 %
     epoch 7, minibatch 2500/2500, test error of best model 2.700000 %
momentum:  0.565937280655
learning rate:  0.0899608631008
epoch 8, minibatch 2500/2500, validation error 2.400000 %
     epoch 8, minibatch 2500/2500, test error of best model 2.430000 %
momentum:  0.574618518353
learning rate:  0.0886114501543
epoch 9, minibatch 2500/2500, validation error 2.120000 %
     epoch 9, minibatch 2500/2500, test error of best model 2.440000 %
momentum:  0.58312612772
learning rate:  0.0872822784019
epoch 10, minibatch 2500/2500, validation error 2.300000 %
momentum:  0.591463625431
learning rate:  0.0859730442259
epoch 11, minibatch 2500/2500, validation error 2.140000 %
momentum:  0.599634349346
learning rate:  0.0846834485625
epoch 12, minibatch 2500/2500, validation error 2.220000 %
momentum:  0.607641637325
learning rate:  0.0834131968341
epoch 13, minibatch 2500/2500, validation error 2.180000 %
momentum:  0.615488827229
learning rate:  0.0821619988816
epoch 14, minibatch 2500/2500, validation error 2.200000 %
momentum:  0.623179078102
learning rate:  0.0809295688984
epoch 15, minibatch 2500/2500, validation error 2.110000 %
     epoch 15, minibatch 2500/2500, test error of best model 2.250000 %
momentum:  0.630715489388
learning rate:  0.0797156253649
epoch 16, minibatch 2500/2500, validation error 2.140000 %
momentum:  0.638101160526
learning rate:  0.0785198909844
epoch 17, minibatch 2500/2500, validation error 2.050000 %
     epoch 17, minibatch 2500/2500, test error of best model 2.070000 %
momentum:  0.645339131355
learning rate:  0.0773420926196
epoch 18, minibatch 2500/2500, validation error 2.000000 %
     epoch 18, minibatch 2500/2500, test error of best model 2.010000 %
momentum:  0.652432322502
learning rate:  0.0761819612303
epoch 19, minibatch 2500/2500, validation error 2.230000 %
momentum:  0.659383654594
learning rate:  0.0750392318119
epoch 20, minibatch 2500/2500, validation error 1.880000 %
     epoch 20, minibatch 2500/2500, test error of best model 2.020000 %
momentum:  0.666195988655
learning rate:  0.0739136433347
epoch 21, minibatch 2500/2500, validation error 2.170000 %
momentum:  0.672872066498
learning rate:  0.0728049386847
epoch 22, minibatch 2500/2500, validation error 1.900000 %
momentum:  0.679414629936
learning rate:  0.0717128646044
epoch 23, minibatch 2500/2500, validation error 2.090000 %
momentum:  0.685826361179
learning rate:  0.0706371716354
epoch 24, minibatch 2500/2500, validation error 2.000000 %
momentum:  0.692109823227
learning rate:  0.0695776140608
epoch 25, minibatch 2500/2500, validation error 1.950000 %
momentum:  0.698267638683
learning rate:  0.0685339498499
epoch 26, minibatch 2500/2500, validation error 1.970000 %
momentum:  0.704302310944
learning rate:  0.0675059406022
epoch 27, minibatch 2500/2500, validation error 1.930000 %
momentum:  0.710216283798
learning rate:  0.0664933514931
epoch 28, minibatch 2500/2500, validation error 1.940000 %
momentum:  0.716011941433
learning rate:  0.0654959512207
epoch 29, minibatch 2500/2500, validation error 1.950000 %
momentum:  0.721691727638
learning rate:  0.0645135119524
epoch 30, minibatch 2500/2500, validation error 1.840000 %
     epoch 30, minibatch 2500/2500, test error of best model 1.900000 %
momentum:  0.727257907391
learning rate:  0.0635458092731
epoch 31, minibatch 2500/2500, validation error 2.030000 %
momentum:  0.732712745667
learning rate:  0.062592622134
epoch 32, minibatch 2500/2500, validation error 1.880000 %
momentum:  0.738058507442
learning rate:  0.061653732802
epoch 33, minibatch 2500/2500, validation error 1.800000 %
     epoch 33, minibatch 2500/2500, test error of best model 2.010000 %
momentum:  0.743297338486
learning rate:  0.06072892681
epoch 34, minibatch 2500/2500, validation error 1.960000 %
momentum:  0.748431384563
learning rate:  0.0598179929078
epoch 35, minibatch 2500/2500, validation error 2.010000 %
momentum:  0.753462731838
learning rate:  0.0589207230142
epoch 36, minibatch 2500/2500, validation error 1.820000 %
momentum:  0.758393466473
learning rate:  0.058036912169
epoch 37, minibatch 2500/2500, validation error 1.950000 %
momentum:  0.763225615025
learning rate:  0.0571663584865
epoch 38, minibatch 2500/2500, validation error 1.960000 %
momentum:  0.767961084843
learning rate:  0.0563088631092
epoch 39, minibatch 2500/2500, validation error 1.850000 %
momentum:  0.77260184288
learning rate:  0.0554642301625
epoch 40, minibatch 2500/2500, validation error 2.030000 %
momentum:  0.777149796486
learning rate:  0.0546322667101
epoch 41, minibatch 2500/2500, validation error 1.910000 %
momentum:  0.781606793404
learning rate:  0.0538127827095
epoch 42, minibatch 2500/2500, validation error 1.920000 %
momentum:  0.785974681377
learning rate:  0.0530055909688
epoch 43, minibatch 2500/2500, validation error 1.940000 %
momentum:  0.790255188942
learning rate:  0.0522105071043
epoch 44, minibatch 2500/2500, validation error 1.910000 %
momentum:  0.794450104237
learning rate:  0.0514273494977
epoch 45, minibatch 2500/2500, validation error 1.790000 %
     epoch 45, minibatch 2500/2500, test error of best model 1.880000 %
momentum:  0.798561096191
learning rate:  0.0506559392553
epoch 46, minibatch 2500/2500, validation error 1.820000 %
momentum:  0.802589893341
learning rate:  0.0498961001664
epoch 47, minibatch 2500/2500, validation error 1.840000 %
momentum:  0.806538105011
learning rate:  0.0491476586639
epoch 48, minibatch 2500/2500, validation error 1.850000 %
momentum:  0.810407340527
learning rate:  0.048410443784
epoch 49, minibatch 2500/2500, validation error 1.860000 %
momentum:  0.814199209213
learning rate:  0.0476842871272
epoch 50, minibatch 2500/2500, validation error 2.010000 %
momentum:  0.817915201187
learning rate:  0.0469690228203
epoch 51, minibatch 2500/2500, validation error 1.930000 %
momentum:  0.821556925774
learning rate:  0.046264487478
epoch 52, minibatch 2500/2500, validation error 1.710000 %
     epoch 52, minibatch 2500/2500, test error of best model 1.870000 %
momentum:  0.825125813484
learning rate:  0.0455705201658
epoch 53, minibatch 2500/2500, validation error 1.840000 %
momentum:  0.82862329483
learning rate:  0.0448869623633
epoch 54, minibatch 2500/2500, validation error 2.010000 %
momentum:  0.832050800323
learning rate:  0.0442136579279
epoch 55, minibatch 2500/2500, validation error 1.730000 %
momentum:  0.835409760475
learning rate:  0.043550453059
epoch 56, minibatch 2500/2500, validation error 1.870000 %
momentum:  0.838701546192
learning rate:  0.0428971962631
epoch 57, minibatch 2500/2500, validation error 1.780000 %
momentum:  0.841927528381
learning rate:  0.0422537383191
epoch 58, minibatch 2500/2500, validation error 1.770000 %
momentum:  0.84508895874
learning rate:  0.0416199322444
epoch 59, minibatch 2500/2500, validation error 1.860000 %
momentum:  0.848187208176
learning rate:  0.0409956332607
epoch 60, minibatch 2500/2500, validation error 1.960000 %
momentum:  0.851223468781
learning rate:  0.0403806987618
epoch 61, minibatch 2500/2500, validation error 1.960000 %
momentum:  0.854198992252
learning rate:  0.0397749882803
epoch 62, minibatch 2500/2500, validation error 2.070000 %
momentum:  0.857115030289
learning rate:  0.0391783634561
epoch 63, minibatch 2500/2500, validation error 1.770000 %
momentum:  0.859972715378
learning rate:  0.0385906880043
epoch 64, minibatch 2500/2500, validation error 1.970000 %
momentum:  0.862773239613
learning rate:  0.0380118276842
epoch 65, minibatch 2500/2500, validation error 2.000000 %
momentum:  0.865517795086
learning rate:  0.037441650269
epoch 66, minibatch 2500/2500, validation error 2.160000 %
momentum:  0.868207454681
learning rate:  0.0368800255149
epoch 67, minibatch 2500/2500, validation error 1.930000 %
momentum:  0.870843291283
learning rate:  0.0363268251322
epoch 68, minibatch 2500/2500, validation error 2.000000 %
momentum:  0.873426437378
learning rate:  0.0357819227552
epoch 69, minibatch 2500/2500, validation error 1.820000 %
momentum:  0.875957906246
learning rate:  0.0352451939139
epoch 70, minibatch 2500/2500, validation error 1.790000 %
momentum:  0.878438770771
learning rate:  0.0347165160052
epoch 71, minibatch 2500/2500, validation error 1.880000 %
momentum:  0.880869984627
learning rate:  0.0341957682651
epoch 72, minibatch 2500/2500, validation error 1.990000 %
momentum:  0.883252561092
learning rate:  0.0336828317411
epoch 73, minibatch 2500/2500, validation error 1.960000 %
momentum:  0.885587513447
learning rate:  0.033177589265
epoch 74, minibatch 2500/2500, validation error 1.990000 %
momentum:  0.88787573576
learning rate:  0.032679925426
epoch 75, minibatch 2500/2500, validation error 1.870000 %
momentum:  0.89011824131
learning rate:  0.0321897265447
epoch 76, minibatch 2500/2500, validation error 1.790000 %
momentum:  0.892315864563
learning rate:  0.0317068806465
epoch 77, minibatch 2500/2500, validation error 1.990000 %
momentum:  0.894469559193
learning rate:  0.0312312774368
epoch 78, minibatch 2500/2500, validation error 2.310000 %
momentum:  0.896580159664
learning rate:  0.0307628082752
epoch 79, minibatch 2500/2500, validation error 1.820000 %
momentum:  0.898648560047
learning rate:  0.0303013661511
epoch 80, minibatch 2500/2500, validation error 1.870000 %
momentum:  0.900675594807
learning rate:  0.0298468456588
epoch 81, minibatch 2500/2500, validation error 1.640000 %
     epoch 81, minibatch 2500/2500, test error of best model 1.790000 %
momentum:  0.902662098408
learning rate:  0.029399142974
epoch 82, minibatch 2500/2500, validation error 1.800000 %
momentum:  0.904608845711
learning rate:  0.0289581558294
epoch 83, minibatch 2500/2500, validation error 1.870000 %
momentum:  0.906516671181
learning rate:  0.0285237834919
epoch 84, minibatch 2500/2500, validation error 1.810000 %
momentum:  0.908386349678
learning rate:  0.0280959267395
epoch 85, minibatch 2500/2500, validation error 2.050000 %
momentum:  0.910218596458
learning rate:  0.0276744878384
epoch 86, minibatch 2500/2500, validation error 1.900000 %
momentum:  0.912014245987
learning rate:  0.0272593705209
epoch 87, minibatch 2500/2500, validation error 1.820000 %
momentum:  0.913773953915
learning rate:  0.026850479963
epoch 88, minibatch 2500/2500, validation error 2.030000 %
momentum:  0.915498495102
learning rate:  0.0264477227636
epoch 89, minibatch 2500/2500, validation error 1.740000 %
momentum:  0.9171885252
learning rate:  0.0260510069221
epoch 90, minibatch 2500/2500, validation error 1.950000 %
momentum:  0.918844759464
learning rate:  0.0256602418183
epoch 91, minibatch 2500/2500, validation error 1.740000 %
momentum:  0.920467853546
learning rate:  0.025275338191
epoch 92, minibatch 2500/2500, validation error 1.930000 %
momentum:  0.922058522701
learning rate:  0.0248962081182
epoch 93, minibatch 2500/2500, validation error 1.890000 %
momentum:  0.923617362976
learning rate:  0.0245227649964
epoch 94, minibatch 2500/2500, validation error 1.870000 %
momentum:  0.925145030022
learning rate:  0.0241549235215
epoch 95, minibatch 2500/2500, validation error 1.870000 %
momentum:  0.926642119884
learning rate:  0.0237925996686
epoch 96, minibatch 2500/2500, validation error 1.740000 %
momentum:  0.928109288216
learning rate:  0.0234357106736
epoch 97, minibatch 2500/2500, validation error 1.900000 %
momentum:  0.929547131062
learning rate:  0.0230841750135
epoch 98, minibatch 2500/2500, validation error 1.730000 %
momentum:  0.930956184864
learning rate:  0.0227379123883
epoch 99, minibatch 2500/2500, validation error 1.870000 %
momentum:  0.93233704567
learning rate:  0.0223968437025
epoch 100, minibatch 2500/2500, validation error 1.880000 %
Optimization complete. Best validation score of 1.640000 % obtained at iteration 202500, with test performance 1.790000 %
train_set_x shape :
In [ ]:


In [1]:
from mlp_dropout_dropconnect import *

for p in [0.6,0.7,0.8,0.9]:
    test_mlp(learning_rate = 0.1, momentum = 0.5, L1_reg= 0.00, L2_reg= 0.0001, n_epochs=100, batch_size=20, n_hidden=800, n_hiddenLayers=2, hidden_layer_type= DropoutHiddenLayer,p = p, dataset = 'MNIST', verbose = True, smaller_set = False,filename = 'mlp_dropout_mnist_linear_%s'%p,activation = linear_rectifier)

train_set_x shape : (50000, 784)
test_set_x shape : (10000, 784)
valid_set_x shape : (10000, 784)
n_train_batches : 2500
n_test_batches : 500
n_valid_batches : 500
... building the model
Model Build
... training
momentum:  0.5
learning rate:  0.1
epoch 1, minibatch 2500/2500, validation error 5.760000 %
     epoch 1, minibatch 2500/2500, test error of best model 6.580000 %
momentum:  0.509999990463
learning rate:  0.0985
epoch 2, minibatch 2500/2500, validation error 3.520000 %
     epoch 2, minibatch 2500/2500, test error of best model 4.050000 %
momentum:  0.519800007343
learning rate:  0.0970225
epoch 3, minibatch 2500/2500, validation error 3.280000 %
     epoch 3, minibatch 2500/2500, test error of best model 3.620000 %
momentum:  0.529403984547
learning rate:  0.0955671625
epoch 4, minibatch 2500/2500, validation error 2.760000 %
     epoch 4, minibatch 2500/2500, test error of best model 2.900000 %
momentum:  0.538815915585
learning rate:  0.0941336550625
epoch 5, minibatch 2500

Using gpu device 0: GRID K520
